src/ts/services/keyring-vault.ts

```diff
@@ -0,0 +1,171 @@
+    if (this.isUnlocked()) throw new Error(t('error.wallet.unlocked'))
+    if (!password.length) throw new Error(t('error.password'))
+    // this will be redundant if we have polkadot js api initialisation
+    return cryptoWaitReady().then(async () => {
+      this._keyring = new Keyring({ addressPrefix, type: 'sr25519' })
+      let vault = await LocalStore.get(this._vaultKey)
+      if (vault) {
+        let accounts = Object.values(vault[this._vaultKey])
+        try {
+          accounts.forEach((account) => {
+            let pair = this.keyring.addFromJson(account as KeyringPair$Json)
+            pair.decodePkcs8(password)
+            this.keyring.addPair(pair)
+          })
+          this._password = password
+          return accounts as Array<KeyringPair$Json>
+        } catch (e) {
+          this.keyring.getPairs().forEach((pair) => {
+            this.keyring.removePair(pair.address())
+          })
+          throw new Error(t('error.password'))
+        }
+      }
+      this._password = password
+      return []
+    })
+  }
+
+  getAccounts (): Array<KeyringPair$Json> {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    let accounts = new Array<KeyringPair$Json>()
+    this.keyring.getPairs().forEach((pair) => {
+      accounts.push(pair.toJson(this._password))
+    })
+    return accounts
+  }
+
+  lock () {
+    this._password = undefined
+    this._mnemonic = undefined
+    this._keyring = undefined
+  }
+
+  generateMnemonic (): string {
+    if (this.isLocked()) {
+      throw new Error(t('error.wallet.locked'))
+    }
+    this._mnemonic = mnemonicGenerate()
+    return this._mnemonic
+  }
+
+  createAccount (mnemonic: string, accountName: string): Promise<KeyringPair$Json> {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    if (this._mnemonic !== mnemonic) throw new Error(t('error.mnemonic.unmatched'))
+    let pair = this.keyring.addFromMnemonic(mnemonic, { name: accountName })
+    this._mnemonic = undefined
+    return this.saveAccount(pair).then((keyringPair$Json: KeyringPair$Json) => {
+      return keyringPair$Json
+    })
+  }
+
+  updateAccountName (address: string, accountName: string): Promise<KeyringPair$Json> {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    const pair = this.keyring.getPair(address)
+    if (!pair) throw new Error(t('error.account.notFound'))
+    let meta = pair.getMeta()
+    meta.name = accountName
+    pair.setMeta(meta)
+    this.keyring.addPair(pair)
+    return this.saveAccount(pair).then((json) => {
+      return json
+    })
+  }
+
+  removeAccount (address: string) {
+    if (this.isLocked()) throw new Error(t('error.wallet.locked'))
+    this.keyring.removePair(address)
+    LocalStore.get(this._vaultKey).then((vault) => {
+      if (vault && vault[this._vaultKey]) {
+        delete vault[this._vaultKey][address]
+        return LocalStore.set(vault[this._vaultKey])
+      }
+      return
```

Comment:
dont need return here

Reply:
Typescript complains if we don't return here

Reply:
we probably can add :void in func. return will return `undefined`

Reply:
removed all the returns